src/pk_compiler.c

```diff
@@ -663,8 +670,20 @@ static void eatNumber(Compiler* compiler) {
         eatChar(compiler);
     }
 
+    // parse if in scientific notation format (MeN == M * 10 ** N)
+    if (peekChar(compiler) == 'e' || peekChar(compiler) == 'E') {
+      eatChar(compiler);
+      if (!utilIsDigit(peekChar(compiler))) {
```

Comment:
Here we need to allow a `+` and `-` character for literals like `1.2e-3`

Reply:
fixed